packages/language-server/src/xml-view-diagnostics.ts

```diff
@@ -0,0 +1,109 @@
+/* istanbul ignore file - Shahar: Tests will be done in a separate PR (exploring snapshot tests) */
+import { map, drop } from "lodash";
+import { assertNever } from "assert-never";
+import {
+  Diagnostic,
+  DiagnosticSeverity,
+  DiagnosticTag
+} from "vscode-languageserver-types";
+import { TextDocument } from "vscode-languageserver-textdocument";
+import { DocumentCstNode, parse } from "@xml-tools/parser";
+import { buildAst } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import {
+  UI5XMLViewIssue,
+  validateXMLView,
+  XMLViewIssueSeverity
+} from "@ui5-language-assistant/xml-views-validation";
+
+export function getXMLViewDiagnostics(opts: {
+  document: TextDocument;
+  ui5Model: UI5SemanticModel;
+}): Diagnostic[] {
+  const documentText = opts.document.getText();
+  const { cst, tokenVector } = parse(documentText);
+  const xmlDocAst = buildAst(cst as DocumentCstNode, tokenVector);
+  const issues = validateXMLView({
+    xmlView: xmlDocAst,
+    model: opts.ui5Model
+  });
+  const diagnostics = validationIssuesToLspDiagnostics(issues, opts.document);
+  return diagnostics;
+}
+
+function validationIssuesToLspDiagnostics(
+  issues: UI5XMLViewIssue[],
+  document: TextDocument
+): Diagnostic[] {
+  const diagnostics: Diagnostic[] = map(issues, currIssue => {
+    const commonDiagnosticPros: Diagnostic = {
+      range: {
+        start: document.positionAt(currIssue.offsetRanges[0].start),
```

Comment:
Shouldn't there be an offset-by-1 issue with the start and end? Since Chevrotain offest is 1-based and VSCode is 0-based

Reply:
offset is 0 based for both, only the line/columns are 1 based in Chevrotain